massgov/mayflower
# DP-16312 [a11y] Campaign landing page key message

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

Apply the accessibility remediation to resolve the mark up validation issues for the key message component.

- Reposition <style> and <script> for the background image above the key message container.
- Remove the extra curly brace causing a parse error.
- Add a condition to add ID only when its value is available to key message section.
- Add a condition to button link to add title only when its value is available.
- Modify the condition to add the second set of `<style>` to limit to the header one only.

No visual change.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-16312)

## Steps to Test


1. Go to the sample page:  /patterns/05-pages-marketing-campaign/05-pages-marketing-campaign.html.
2. Using Inspect in the browser, navigate to the header key message.
3. Find `<style>` and `<script>` for the background image locates above the key message container.
<img width="645" alt="Screen Shot 2019-11-01 at 12 00 02 PM" src="https://user-images.githubusercontent.com/9633303/68037997-53f3d880-fc9f-11e9-8732-c12e088f9593.png">
4. Check the markup for the button and find no title attribute.

Before
<img width="881" alt="before" src="https://user-images.githubusercontent.com/9633303/68039251-1e042380-fca2-11e9-9b32-b07da41c7606.png">
After
<img width="856" alt="after" src="https://user-images.githubusercontent.com/9633303/68039261-24929b00-fca2-11e9-9748-7e8feabb8065.png">

5. Go to /patternlab/styleguide/source/_patterns/05-pages/marketing-campaign.json.
6. Copy L. 11("bgImage": "../../assets/images/placeholder/1920x1080-sign.jpg",) and paste it after L.80.  As the sample page gets refreshed, find a background image is displayed in the key message right above the footer.
7. In the Inspect, locate the key message.  Find it has only one set of `<style>` as shown in the **after** screenshots below.  Check its mobile presentation for any difference between the current implementation with 2 `<style>`s.

**After - Desktop**
<img width="744" alt="style-after-desktop" src="https://user-images.githubusercontent.com/9633303/68134694-7d577300-fef0-11e9-8462-c7f048f53ef7.png">

**After - Mobile**
<img width="700" alt="style-after-mobile" src="https://user-images.githubusercontent.com/9633303/68134707-81839080-fef0-11e9-8bd8-8b2f30ec4657.png">

**Before**
<img width="738" alt="style-before" src="https://user-images.githubusercontent.com/9633303/68134633-69137600-fef0-11e9-828d-27509fd040c9.png">



## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
